### PR TITLE
Upgrade Vitess dependency to 13.0.0

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR upgrades the operator dependency to release 13.0.0. As part of this upgrade, some build issues had to be fixed due to refactors in Vitess commands like (https://github.com/vitessio/vitess/pull/9015). 

We have also introduced a new semi-sync parameter in some RPCs, but it is not being used. In order to fix the build issues, we are passing a false value for this parameter. As and when we start using this parameter, we should also revisit this code. The upgrade path that has been decided is to not use this parameter in release-14.0 (making this change safe). We will store the durability_policy in the topo instead of a flag to binaries like vtctld. The RPCs for this have been introduced in https://github.com/vitessio/vitess/pull/10221.

We should use those RPCs in these places, to find the correct durability_policy and use that to set semi-sync. This will be needed to be done when we upgrade the Vitess dependency to release 14.0.0

We have also needed to set a durability_policy in the main() function. Currently we are setting the `none` duraibility`. When we upgrade the Vitess dependency this will also have to change.

### CODE REVIEW

File: pkg/controller/vitessshardreplication/init_restored_shard.go
Comment (deepthi) [inline pkg/controller/vitessshardreplication/init_restored_shard.go:235]: Can you document why we might change this behavior?

File: test/endtoend/operator/101_initial_cluster.yaml
Comment (deepthi) [inline test/endtoend/operator/101_initial_cluster.yaml:11]: should this be 13.0.0 or a later minor version?

File: test/endtoend/operator/101_initial_cluster_backup.yaml
Comment (deepthi) [inline test/endtoend/operator/101_initial_cluster_backup.yaml:18]: why is this latest and not v13.0.x?

**Review by deepthi — COMMENTED** [review]
Nice work!
I like that we have started using VtctldServer.
A few questions, the rest lgtm.
We should get a review from @Verolop before merging.

File: test/endtoend/operator/101_initial_cluster.yaml
Comment (GuptaManan100) [reply test/endtoend/operator/101_initial_cluster.yaml:11]: We can do any/all. I just added this to check that we are able to upgrade from the earliest release

File: test/endtoend/operator/101_initial_cluster_backup.yaml
Comment (GuptaManan100) [reply test/endtoend/operator/101_initial_cluster_backup.yaml:18]: This file is used for the backup test and not the upgrade test. In the backup test we are testing that the backups work with the latest version of Vitess. Earlier I had 13.0.0 there, because the code on the operator was supposed to only work with 13.0.0. After this PR, the code will be meant to be working with the release-14 of Vitess, so we use the latest image.

File: pkg/controller/vitessshardreplication/init_restored_shard.go
Comment (GuptaManan100) [reply pkg/controller/vitessshardreplication/init_restored_shard.go:235]: Do you want me to add it in the code as more comments?

**Review by Verolop — APPROVED** [review]
LGTM!

File: pkg/controller/vitessshardreplication/init_restored_shard.go
Comment (deepthi) [reply pkg/controller/vitessshardreplication/init_restored_shard.go:235]: link to a vitessio/vitess issue if we have one?

**Pull request merged**
